gradle-plugin/src/main/java/com/squareup/hephaestus/plugin/HephaestusPlugin.kt

```diff
@@ -22,26 +24,41 @@ open class HephaestusPlugin : Plugin<Project> {
   @ExperimentalStdlibApi
   override fun apply(project: Project) {
     val once = AtomicBoolean()
-    val innerApply = Action<AppliedPlugin> {
-      if (once.compareAndSet(false, true)) {
-        realApply(project)
+
+    fun PluginManager.withPluginOnce(
+      id: String,
+      action: (AppliedPlugin) -> Unit
+    ) {
+      withPlugin(id) {
+        if (once.compareAndSet(false, true)) {
+          action(it)
+        }
       }
     }
-    with(project.pluginManager) {
-      withPlugin("org.jetbrains.kotlin.android", innerApply)
-      withPlugin("org.jetbrains.kotlin.jvm", innerApply)
+
+    project.pluginManager.withPluginOnce("org.jetbrains.kotlin.android") {
+      realApply(project, true)
     }
+    project.pluginManager.withPluginOnce("org.jetbrains.kotlin.jvm") {
```

Comment:
Can you add a comment on why this is necessary, please ?

Reply:
Done